### PR TITLE
add an action to deploy to github pages when pushed to master.

### LINKED ISSUE #75: Add CI that auto-deploys Sphinx docs after merges to master
It should do something like (now doing locally)
```
# Install packages in requirements.txt
git clean -xdf
pushd spec
make html
popd
(git checkout --orphan tmp && git branch -D gh-pages || true)
git checkout --orphan gh-pages
git reset --hard
cp -R spec/_build/html/ latest
touch .nojekyll
git add .nojekyll
git add latest
git commit -m "Commit API Standard doc build";
git push --set-upstream origin gh-pages --force
```

Put the above in a file `deploy-ghpages.sh` at `..` from the repo root, then run (from `master`, in the repo root):
```
source ../deploy-ghpages.sh
```

### PR DESCRIPTION
**Pull request by tonyfast**
fixes #75

### CODE REVIEW

File: .github/workflows/pages.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+    # - uses: docker://pandoc/latex:2.9
+    - uses: actions/checkout@master
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.8' # Version range or exact version of a Python version to use, using semvers version range syntax.
+        architecture: 'x64' # (x64 or x86)
+    - run: |
+        sudo apt install pandoc
```
Comment (rgommers) [inline .github/workflows/pages.yml:16]: Just a question, did the build not work without `pandoc`? I have it installed but it wasn't in my conda env, so if it's needed it was a very hidden dependency.

File: .github/workflows/pages.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        # add dependencies based on the conf.py
+        pip install sphinx myst-parser sphinx-copybutton sphinx-markdown-tables
```
Comment (rgommers) [inline .github/workflows/pages.yml:18]: `pip install -r `requirements.txt`? shorter and probably more future-proof.

File: .github/workflows/pages.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        sphinx-build spec build
```
Comment (rgommers) [inline .github/workflows/pages.yml:19]: Should be `build/latest` I believe, to deploy to https://data-apis.github.io/array-api/latest/

**Review by rgommers — COMMENTED** [review]
Thanks @tonyfast! Looks great, just a few comments. The last one is the most important, `latest/` needs to be in the URLs to not break incoming links.

File: .github/workflows/pages.yml (hunk 1 above)
Comment (tonyfast) [reply .github/workflows/pages.yml]: i removed this line. we shouldnt need. i think it is necessary with latex.

File: .github/workflows/pages.yml (hunk 2 above)
Comment (tonyfast) [reply .github/workflows/pages.yml:20]: this makes sense. where would you like the requirements to go?

**Comment by tonyfast** [comment]
i resolved your comments. i set up the actions to create an artifact of the documents so they can be previewed. the github pages deployment only updates on changes to master.

File: .github/workflows/pages.yml (hunk 2 above)
Comment (rgommers) [reply .github/workflows/pages.yml:17]: They're already present in the repo root. With version pinning for Sphinx (safer), and `sphinx-material` which is missing here. I'll push a fix.

**Review by rgommers — APPROVED** [review]
Tested on my fork and fixed all remaining issues, let's give this a go! Thanks @tonyfast!

**Pull request merged**